.circleci/config.yml

```diff
@@ -179,6 +192,40 @@ references:
       cp -r ./reports $CIRCLE_TEST_REPORTS/e2ereports &&
       cp -r ./screenshots $CIRCLE_ARTIFACTS/screenshots
 
+  desktop-install-app-deps: &desktop-install-app-deps
+      name: Install Desktop Dependencies
+      command: |
+        source "$HOME/.nvm/nvm.sh"
+        nvm use
+        npm install -g yarn
+        yarn run build-desktop:install-app-deps
+  desktop-cache-paths: &desktop-cache-paths
+    - desktop/build
+    - desktop/client
+    - desktop/config
+    - desktop/public
+    - desktop/resource/certificates/win.p12
+    - desktop/resource/certificates/mac.p12
+    - build
+    - public
+    - client/server
+    - config/secrets.json
+    - client/desktop/config.json
```

Comment:
The purpose of this cache is to save the results of the source build and then restore them for the platform-specific packaging jobs?

For that, you should need only the `desktop/` directory contents. `build/` and `public/` shouldn't be created by any `build-desktop:*` task at all, and all the files in `config/` or `client/` should be either copied to `desktop/` by the `build-desktop:config` task, or generated directly into `desktop/`.

Reply:
> The purpose of this cache is to save the results of the source build and then restore them for the platform-specific packaging jobs?

Yes, this is correct. This was tricky to get right, so I ended up caching all places that generated a diff after the node module installation step. Will try limiting the cache paths to just those within the `desktop` folder. 👍

Reply:
Caching only items in `desktop` works - updated in 8f6dca6.

Reply:
Seems there is first class CircleCI support for sharing files between jobs: the `persist_to_workspace` and `attach_workspace` commands: https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs Can we possibly use these instead of the cache? It can be a followup PR. All my knowledge about this comes from a minute of googling, so I apologize in case I'm suggesting something silly 🙂